src/components/SideNav.jsx

```diff
@@ -1,20 +1,48 @@
 import { NavLink } from "react-router-dom";
+import { useDBdata } from "../context/db-data-context";
 import { useMessageHandling } from "../context/message-handling-context";
 import "../stylesheets/sidenav.css";
 import { sidenavItemList } from "../utilities/sidenav-items";
+import { v4 as uuid } from "uuid";
+
 export function SideNav() {
+  const { labels } = useDBdata();
   const { showSidenav, setShowSidenav } = useMessageHandling();
   return (
     <aside
       className={`sidenav-bg ${!showSidenav ? "sidenav-hide" : ""}`}
       onClick={() => setShowSidenav(false)}
     >
       <ul className="list-group-stacked sidenav">
-        {sidenavItemList.map((item) => (
+        <NavLink
```

Comment:
I will suggest you can make a navlink custom componet with the default active and inactive style you are using , as it will reduce repetititon.